humatest/humatest.go

```diff
@@ -167,27 +192,52 @@ func (a *testAPI) Do(method, path string, args ...any) *httptest.ResponseRecorde
 
 func (a *testAPI) Get(path string, args ...any) *httptest.ResponseRecorder {
 	a.tb.Helper()
-	return a.Do(http.MethodGet, path, args...)
+	return a.GetCtx(context.Background(), path, args...)
+}
+
+func (a *testAPI) GetCtx(ctx context.Context, path string, args ...any) *httptest.ResponseRecorder {
+	a.tb.Helper()
+	return a.DoCtx(ctx, http.MethodGet, path, args...)
 }
 
 func (a *testAPI) Post(path string, args ...any) *httptest.ResponseRecorder {
 	a.tb.Helper()
-	return a.Do(http.MethodPost, path, args...)
+	return a.PostCtx(context.Background(), path, args...)
+}
+
+func (a *testAPI) PostCtx(ctx context.Context, path string, args ...any) *httptest.ResponseRecorder {
+	a.tb.Helper()
+	return a.DoCtx(ctx, http.MethodPost, path, args...)
 }
 
 func (a *testAPI) Put(path string, args ...any) *httptest.ResponseRecorder {
 	a.tb.Helper()
-	return a.Do(http.MethodPut, path, args...)
+	return a.PutCtx(context.Background(), path, args...)
+}
+
+func (a *testAPI) PutCtx(ctx context.Context, path string, args ...any) *httptest.ResponseRecorder {
+	a.tb.Helper()
+	return a.DoCtx(ctx, http.MethodPut, path, args...)
 }
 
 func (a *testAPI) Patch(path string, args ...any) *httptest.ResponseRecorder {
 	a.tb.Helper()
-	return a.Do(http.MethodPatch, path, args...)
+	return a.PatchCtx(context.Background(), path, args...)
```

Comment:
Here and for all existing code you update.

I would expect to see the context defined as a variable, with a comment explaining why you need this

I would also expect the code to add a `//nolint: context check // some explanation`

To avoid any linting issue

As you may duplicate this code and comment, maybe creating a function doing that would be a good thing.

Reply:
I'm not entirely sure I follow this feedback, let me know if I'm misunderstanding the comment. Passing context.Background is pretty common as an argument (for example, [from http](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/net/http/request.go;l=856), or from [this repo](https://github.com/coury-clark/huma/blob/e55b175a0293854df7e43bafe48679667e33322f/examples/sse/main.go#L183)). I don't see any context check linters configured in the repo, and don't see this lint exclusion elsewhere in the repo. 

The linter doesn't seem to mind for me either:
``` 
➜  huma git:(tests-with-ctx) ✗ golangci-lint run
registry.go:140:33: directive `//nolint:musttag` is unused for linter "musttag" (nolintlint)
        return json.Marshal(r.schemas) //nolint:musttag
                                       ^
validate_test.go:1292:67: directive `//nolint: musttag` is unused for linter "musttag" (nolintlint)
                                schemaJSON, _ := json.MarshalIndent(registry.Map(), "", "  ") //nolint: musttag

```

I think this (context.Background()) is a pretty common thing to see, let me know.

Reply:
You can ignore this comment.

I'm a linter fanatic. I tend to write everything as if it would be linted with all option on.